code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1201,7 +1201,7 @@ var/list/rank_prefix = list(\
 				new organ_type(src)
 
 		var/datum/category_item/setup_option/core_implant/I = Pref.get_option("Core implant")
-		if(I.implant_type)
+		if(I.implant_type && (!mind || mind.assigned_role != "Robot"))
```

Comment:
Might want to check for AIs, too.

Reply:
the code to create a body for a new player is a mess and as far as I could test the AIs do not suffer from this problem.